Wikipedia/Code/WMFAppViewController.m

```diff
@@ -1202,8 +1201,28 @@ - (BOOL)processUserActivity:(NSUserActivity *)activity animated:(BOOL)animated c
                 done();
                 return NO;
             }
-            [self showArticleForURL:URL animated:animated completion:done];
-            // don't call done block before this return, wait for completion ^
+            [[WMFSession shared] fetchSummaryForArticleURL:URL priority:NSURLSessionTaskPriorityHigh completionHandler:^(NSDictionary<NSString *,id> * _Nullable result, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    dispatch_block_t bail = ^{
+                        [self wmf_openExternalUrl:URL];
+                    };
+                    NSDictionary *namespaceDictionary = result[@"namespace"];
+                    if (![namespaceDictionary isKindOfClass:[NSDictionary class]]) {
+                        bail();
+                        return;
+                    }
+                    NSNumber *namespaceId = namespaceDictionary[@"id"];
+                    if (![namespaceId isKindOfClass:[NSNumber class]]) {
+                        bail();
+                        return;
+                    }
+                    if ([namespaceId integerValue] != 0) {
+                        bail();
+                        return;
+                    }
+                    [self showArticleForURL:URL animated:animated completion:done];
```

Comment:
wouldn't you want to `bail` if there's an error?

Reply:
When there's an error, `response` will be nil and it'll bail at the first check

Reply:
@nambatee I thought about this some more, the app should show the error in this case. If it's a network error or some other error, pushing into the SFSafariVC would just be confusing. I added this change.